src/plugins/php/templates/lr.template.php

```diff
@@ -105,6 +105,10 @@ public static function parse($string) {
 
       $state = end($stack);
       $column = $tokens[$token['type']];
+
+      if (!isset($table[$state][$column])) {
+          self::unexpectedEndOfInput();
```

Comment:
Good catch! But let's throw `unexpectedToken` instead here. This check already done below with:

```php
if (!$entry) {
  self::unexpectedToken($token);
}
```

So we can just change this to:

```php
if (!isset($table[$state][$column])) {
  self::unexpectedToken($token);
}

$entry = $table[$state][$column];
```

And then remove `if (!$entry) {` check below.

nit: 2 spaces indent